be/src/olap/base_compaction.cpp

```diff
@@ -40,487 +39,264 @@ using std::vector;
 
 namespace doris {
 
-OLAPStatus BaseCompaction::init(TabletSharedPtr tablet, bool is_manual_trigger) {
-    // 表在首次查询或PUSH等操作时，会被加载到内存
-    // 如果表没有被加载，表明该表上目前没有任何操作，所以不进行BE操作
-    if (!tablet->init_succeeded()) {
-        return OLAP_ERR_INPUT_PARAMETER_ERROR;
-    }
-
-    LOG(INFO) << "init base compaction handler. [tablet=" << tablet->full_name() << "]";
+static bool rowset_comparator(const RowsetSharedPtr& left, const RowsetSharedPtr& right) {
+    return left->end_version() < right->end_version();
```

Comment:
Should add a parent class and move all common methods and methods only used by compaction to parent class. There are too many duplicate methods.